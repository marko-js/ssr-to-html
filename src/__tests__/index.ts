import fs from "fs";
import path from "path";
import crawl from "../index";

test("works", async () => {
  const getFiles = trackFS();
  await crawl({
    out: "/",
    cmd: nodeCMD("fixtures/simple-server.js")
  });
  expect(getFiles()).toMatchSnapshot();
});

function trackFS() {
  const {
    createWriteStream,
    promises: { mkdir }
  } = fs;
  const files: Record<string, string> = {};

  fs.promises.mkdir = ((() => Promise.resolve()) as unknown) as typeof mkdir;
  fs.createWriteStream = (((path: string) => {
    files[path] = "";
    return {
      write(data: string | Buffer) {
        files[path] += data.toString("utf-8");
      },
      end(data?: string | Buffer | undefined) {
        if (data) {
          this.write(data);
        }
      }
    };
  }) as unknown) as typeof createWriteStream;

  return () => {
    fs.createWriteStream = createWriteStream;
    fs.promises.mkdir = mkdir;
    return files;
  };
}

function nodeCMD(req: string) {
  return `node ${path.relative(process.cwd(), path.resolve(__dirname, req))}`;
}
