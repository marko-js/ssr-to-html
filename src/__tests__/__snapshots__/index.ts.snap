// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
Object {
  "/404/index.html": "<!DOCTYPE html><h1>404",
  "/bye/index.html": "<!DOCTYPE html><h1>Bye!",
  "/index.html": "<!DOCTYPE html><h1>Hi! <a href='/bye'>Bye!",
}
`;
